File: backend/models/intervention.js
const mongoose = require("mongoose");

const InterventionSchema = new mongoose.Schema({
  _id: Number,
  cptCode: {
    type: Number,
    required: true,
  },
  description: {
    type: String,
    required: true,
  },
  price: {
    type: Number,
    required: true,
  },
});

InterventionSchema.pre("save", function () {
  this._id = this.cptCode;
});

module.exports = mongoose.model("Intervention", InterventionSchema);

File: frontend/lib/infrastructure/patient_records/models/intervention.dart
// ignore_for_file: file_names

class Intervention {
  final int? id;
  final int cptCode;
  final String? description;
  final int? price;

  Intervention({
    required this.id,
    required this.cptCode,
    required this.price,
    this.description
  });

  factory Intervention.fromJson(Map<String, dynamic> json) {
    return Intervention(
        id: json['_id'], 
        cptCode: json['cptCode'], 
        price: json['price'],
        description: json['description']);
  }
}
